src/rust-2024/match-ergonomics.md

```diff
@@ -1,9 +1,135 @@
-# Match ergonomics
-
-**This is a placeholder, docs coming soon!**
+# Match ergonomics reservations
 
 ## Summary
 
+- Writing `mut`, `ref`, or `ref mut` on a binding is only allowed within a pattern when the pattern leading up to that binding is fully explicit (i.e. when it does not use match ergonomics).
+  - Put differently, when the default binding mode is not `move`, writing `mut`, `ref`, or `ref mut` on a binding is an error.
+- Reference patterns (`&` or `&mut`) are only allowed within the fully-explicit prefix of a pattern.
+  - Put differently, reference patterns can only match against references in the scrutinee when the default binding mode is `move`.
```

Comment:
If this seems a bit dense, here and elsewhere throughout the chapter.... sorry.  I tried.  Match ergonomics is just hard.